sql/parser/eval.go

```diff
@@ -136,13 +139,13 @@ var binOps = map[binArgs]binOp{
 	binArgs{Plus, dateType, intervalType}: {
 		returnType: DummyTimestamp,
 		fn: func(left Datum, right Datum) (Datum, error) {
-			return DTimestamp{Time: left.(DDate).Add(right.(DInterval).Duration)}, nil
+			return DTimestamp{Time: left.(DDate).toTime().Add(right.(DInterval).Duration)}, nil
```

Comment:
`toTime` always uses UTC; a date plus or minus an interval or timestamp should be computed in the session time zone. (and we need some tests that expose this)

`toTime` is rarely useful and it looks like we only ever use it correctly when we want to pull the (year, month, day) fields out of a value so we should probably replace `toTime()` with a `toDate() (int, int, int)` method.